packages/core/src/renderers/systems/geometry/GeometrySystem.js

```diff
@@ -137,6 +137,23 @@ export default class GeometrySystem extends WebGLSystem
             {
                 this.activateVao(geometry, shader.program);
             }
+
+            // Currently GeometrySystem is the only system that created element array buffers,
+            // it does not mess with index buffers of different vao's,
+            // but we have no guarantee that some other lib (by @mrdoob)
+            // is not managing the same context
+            //
+            // Pixi is the only lib that treats indexBuffer as a part of vao so far.
+
+            if (geometry.indexBuffer)
+            {
+                const glIndex = geometry.indexBuffer._glBuffers[this.CONTEXT_UID];
+
+                if (glIndex)
+                {
+                    gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, glIndex.buffer);
+                }
+            }
```

Comment:
Does a native VAO not save the bound index buffer as well as any vertex buffers? I though it captured both :)

Reply:
Index buffer is not bound to an attribute, it changes whenever you want to do any operations with that slot.

Other renderers dont respect that. We also didnt respect it in v4.